x-pack/plugins/rule_registry/server/alert_data_client/alerts_client.ts

```diff
@@ -1139,6 +1143,19 @@ export class AlertsClient {
       size: 0,
       _source: false,
     });
+    // Replace artificial uuid values with '--' in null-value buckets and mark them with `isNullGroup = true`
+    const groupsAggregation = searchResult.aggregations
+      ?.groupByFields as AggregationsMultiBucketAggregateBase<{ key: string }>;
```

Comment:
Can we somehow avoid casting here?

Reply:
Done! I added a generic type for aggregations to all the methods down to the esQuery itself 🙂